Source/AviationWeather/Source/decoders.cpp

```
@@ -0,0 +1,214 @@
+    }
+    else if (symbol == "GS") {
+        return weather_phenomena::small_hail;
+    }
+    else if (symbol == "UP") {
+        return weather_phenomena::small_hail;
+    }
+    else if (symbol == "BR") {
+        return weather_phenomena::mist;
+    }
+    else if (symbol == "FG") {
+        return weather_phenomena::fog;
+    }
+    else if (symbol == "FU") {
+        return weather_phenomena::smoke;
+    }
+    else if (symbol == "VA") {
+        return weather_phenomena::volcanic_ash;
+    }
+    else if (symbol == "DU") {
+        return weather_phenomena::widespread_dust;
+    }
+    else if (symbol == "SA") {
+        return weather_phenomena::sand;
+    }
+    else if (symbol == "HZ") {
+        return weather_phenomena::haze;
+    }
+    else if (symbol == "PY") {
+        return weather_phenomena::spray;
+    }
+    else if (symbol == "PO") {
+        return weather_phenomena::well_developed_dust_whirls;
+    }
+    else if (symbol == "SQ") {
+        return weather_phenomena::squalls;
+    }
+    else if (symbol == "FC") {
+        return weather_phenomena::funnel_cloud_tornado_waterspout;
+    }
+    else if (symbol == "SS") {
+        return weather_phenomena::sandstorm;
+    }
+    else if (symbol == "DS") {
+        return weather_phenomena::duststorm;
+    }
+    else {
+        return weather_phenomena::none;
+    }
+}
+
+//-----------------------------------------------------------------------------
+
+sky_cover_type decode_sky_cover(std::string const& symbol)
+{
+    if (symbol == "VV") {
+        return sky_cover_type::vertical_visibility;
+    }
+    else if (symbol == "FEW") {
+        return sky_cover_type::few;
+    }
+    else if (symbol == "SCT") {
+        return sky_cover_type::scattered;
+    }
+    else if (symbol == "BKN") {
+        return sky_cover_type::broken;
+    }
+    else if (symbol == "OVC") {
+        return sky_cover_type::overcast;
+    }
+    else {
+        return sky_cover_type::sky_clear;
```

This should throw an exception if you try to decode something that we don't have an enum for. I'll check the other decoders too.